### ISSUE TITLE
Connection closed before receiving a handshake response in react

### ISSUE THREAD

**Issue by guitar9** [mqttjs/MQTT.js#976]
I have running the mosquitto MQTT Broker in a docker container on localhost:1883 and I get this error message: Connection closed before receiving a handshake response

I am using a react app an started with create-react-app. My WebApp is running on port 3000.
The connection to mqtt://test.mosquitto.org:8080 works fine
My mosquitto.conf:

persistence true
persistence_location /mosquitto/data/
log_dest file /mosquitto/log/mosquitto.log

**Comment by yoshegg** [comment]
I get the same error message since yesterday, after my react app worked for weeks without an issue.

**Comment by ankitmhn** [comment]
Facing the same error. I have a MQTT broker running on localhost:1883 which throws this error. Strangely enough, the app was able to connect to mqtt://test.mosquitto.org:8080 exactly once before this error started showing up for this as well.

**Comment by HADDADSOHAIB** [comment]
Hi, any news about this issue. :upside_down_face:  I am using IOT hub on scaleway.com and connected successfully using the mqttjs package with some js script. But when I tried to do the same inside a react app (created with create-react-app) I got the same error.
![Screenshot from 2020-06-25 01-25-42](https://user-images.githubusercontent.com/26064042/85640142-e4de5080-b682-11ea-9b1e-cb82c5d97ff7.png)
